plugins/inputs/traefik/README.md

```diff
@@ -0,0 +1,87 @@
+# Telegraf Input Plugin: Traefik
+
+This plugin gather health check status from services registered in Traefik. Traefik must be started with the `--api` option (formerly `--web`). See the [configuration docs](https://docs.traefik.io/configuration/api/) for more information.
+
+### Configuration:
+
+Simple configuration, with one measurement per request capture:
+
+```toml
+# Description
+[[inputs.traefik]]
+  ## Required Traefik server address, host and port (default: "127.0.0.1")
+  # address = "http://127.0.0.1:8080"
+  #
+  # [inputs.traefik.tags]
+  #  instance = "prod"
+```
+
+Configuration with additional measurements for each status code encountered:
+
+```toml
+# Description
+[[inputs.traefik]]
+  ## Required Traefik server address, host and port (default: "127.0.0.1")
+  # address = "http://127.0.0.1:8080"
+  # include_status_code_measurement = true
+  #
+  # [inputs.traefik.tags]
+  #  instance = "prod"
+```
+
+### Measurements & Fields:
+
+- `traefik` measurement have the following fields:
+    - **total_count** (int) - total number of responses since the application was last restarted
+    - **average_response_time_sec** (float64) - average time (in seconds) of all responses
+    - **total_response_time_sec** (float64) - total time (in seconds) of all responses
+    - **unixtime** (int) - unix timestamp of when this service was last restarted
+    - **uptime_sec** (int) - elapsed time (in seconds) since this service was restarted
+    - **health_response_time_sec** (float64) - round trip time to gather this set of metrics
+    - Http Codes - each status code count since this service was last restarted
+	- **status_code_200** (int)
+	- **status_code_400** (int)
+	- **status_code_500** (int)
+	- *status_code_*...
+
+- `traefik_status_codes` measurement have the following fields:
```

Comment:
The cardinality of this measurement isn't going to be much of a problem, in almost all cases we are talking several dozen codes.  Even 1000 series wouldn't really be a problem, so I wouldn't make it configurable, keep it simple and always send the status code metrics.

So with that in mind, and given the shared data between the two measurements, I think a single measurement would work okay if we did it similar to:
```
traefik,source=localhost total_count=42i,average_response_time_ns=42i
traefik,source=localhost,status_code=400 status_code_count=42i
traefik,source=localhost,status_code=500 status_code_count=42i
```
If we do it like this, I don't mind having them all under the same measurement because there is no overlap between the field names.  However if you prefer, two measurements are still okay but we should keep the shared fields on the primary measurement only:

```
traefik,source=localhost total_count=42i,average_response_time_ns=42i
traefik_status_code,source=localhost,status_code=400 count=42i
traefik_status_code,source=localhost,status_code=500 count=42i
```